### PR TITLE
reference docs for KafkaSourceProvider

### PR DESCRIPTION
**Pull request by patriknw**
On top of https://github.com/akka/akka-projection/pull/131
Also requires https://github.com/akka/akka-projection/pull/132 to be able to run the example.

* also moved MergableOffset to public place

I intend to add publishing to Kafka in a separate PR, and also mention committing offsets to Kafka. For the latter I guess we will just refer to Alpakka Kafka documentation since Projections is not adding any value.

References #79

### CODE REVIEW

File: examples/src/test/scala/docs/kafka/KafkaDocExample.scala [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    val sourceProvider: SourceProvider[MergeableOffset[Long], ConsumerRecord[String, String]] =
+      KafkaSourceProvider(system, consumerSettings, Set(topicName))
+    //#sourceProvider
+  }
+
+  object IllustrateExactlyOnce {
+    import IllustrateSourceProvider._
+
+    //#exactlyOnce
+    val dbConfig: DatabaseConfig[H2Profile] = DatabaseConfig.forConfig("akka.projection.slick", system.settings.config)
```
Comment (renatocaval) [inline examples/src/test/scala/docs/kafka/KafkaDocExample.scala:98]:
```suggestion
    val dbConfig: DatabaseConfig[H2Profile] = 
      DatabaseConfig.forConfig("akka.projection.slick", system.settings.config)
```
This line is causing horizontal scrolling (as well as the SlickProjection below)

File: examples/src/test/scala/docs/kafka/KafkaDocExample.scala [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    val projectionId = ProjectionId("WordCount", "wordcount-1")
+    val projection =
+      SlickProjection.exactlyOnce(projectionId, sourceProvider, dbConfig, handler = new WordCountHandler(projectionId))
```
Comment (renatocaval) [inline examples/src/test/scala/docs/kafka/KafkaDocExample.scala:101]:
```suggestion
      SlickProjection.exactlyOnce(
        projectionId, 
        sourceProvider, 
        dbConfig, 
        handler = new WordCountHandler(projectionId))
```

Column alignment make it easier to see what are the pieces we are wiring. And in browser this line is forcing a horizontal scrolling.

**Comment by patriknw** [comment]
> Should we mention anything about the MergeableOffset?

Created issue https://github.com/akka/akka-projection/issues/138

File: docs/src/main/paradox/kafka.md [hunk 3]
```diff
@@ -1,5 +1,23 @@
 # Messages from Kafka
 
+A typical source for Projections is messages from Kafka. Akka Projections has integration with 
+[Alpakka Kafka](https://doc.akka.io/docs/alpakka-kafka/current/), which is described in here.
+
+The @apiddoc[KafkaSourceProvider] uses consumer group assignments from Kafka and can resume from offsets stored in
+a database.
+
+Akka Projections can store the offsets from Kafka in a @ref:[relational DB with Slick](slick.md).
```
Comment (renatocaval) [inline docs/src/main/paradox/kafka.md:9]: While writing the docs for the testkit, I realised that this is problematic. 

Maybe better to use the Cassandra example so the Java and the Scala docs align.

Comment (patriknw) [reply docs/src/main/paradox/kafka.md:9]: Good point. We must use Slick for the consuming and that's why I continued with Slick here. We can change that later though, I'm not planning to write the Java sample today.

**Comment by patriknw** [comment]
it's the logger

**Pull request merged**